components/camel-script/src/main/java/org/apache/camel/builder/script/ScriptBuilder.java

```diff
@@ -481,6 +484,13 @@ protected Object runScript(ScriptEngine engine, Exchange exchange, ScriptContext
         if (result == null) {
             result = engine.get("result");
         }
+        
+        // into jdk8 a list will be mapped to ScriptObjectMirror
+        if (engine instanceof NashornScriptEngine && result != null && result instanceof ScriptObjectMirror) {
```

Comment:
you're welcome, unfortunately no, I was just reading the code to learn about this project so I'm not able to answer you about the OSGi compatibility, sorry.